l1-contracts/contracts/bridge/asset-router/L2AssetRouter.sol

```diff
@@ -135,8 +148,13 @@ contract L2AssetRouter is AssetRouterBase, IL2AssetRouter {
     /// @dev do not rely on this function, it will be deprecated in the future
     /// @param _assetId The asset id of the withdrawn asset
     /// @param _assetData The data that is passed to the asset handler contract
-    function withdraw(bytes32 _assetId, bytes memory _assetData) public override {
-        _withdrawSender(_assetId, _assetData, msg.sender, true);
+    function withdraw(bytes32 _assetId, bytes memory _assetData) public override returns (bytes32) {
+        return _withdrawSender(_assetId, _assetData, msg.sender, true);
+    }
+
+    function withdrawToken(address _l2NativeToken, bytes memory _assetData) public returns (bytes32) {
```

Comment:
What will happen if someone provides an address of the token that is not native? I.e. an address for the bridged asset

Reply:
I will add an extra check